runtime/compiler/optimizer/J9Inliner.cpp

```diff
@@ -1167,13 +1169,24 @@ void TR_InlinerBase::getBorderFrequencies(int32_t &hotBorderFrequency, int32_t &
          }
       }
 
-   // Did the user specify specific values? If so, use those
-   if (comp()->getOptions()->getInlinerBorderFrequency() >= 0)
-      hotBorderFrequency = comp()->getOptions()->getInlinerBorderFrequency();
-   //if (comp()->getOptions()->getInlinerColdBorderFrequency() >= 0)
-   //   coldBorderFrequency = comp()->getOptions()->getInlinerColdBorderFrequency();
-   if (comp()->getOptions()->getInlinerVeryColdBorderFrequency() >= 0)
-      coldBorderFrequency = comp()->getOptions()->getInlinerVeryColdBorderFrequency();
+   if (smallerFreqs)
+      {
+      // Did the user specify specific values? If so, use those
+      if (comp()->getOptions()->getSmallInlinerBorderFrequency() >= 0)
+         hotBorderFrequency = comp()->getOptions()->getSmallInlinerBorderFrequency();
+      if (comp()->getOptions()->getSmallInlinerVeryColdBorderFrequency() >= 0)
+         coldBorderFrequency = comp()->getOptions()->getSmallInlinerVeryColdBorderFrequency();
+      }
```

Comment:
I would have put this code on line 1161.
Also, the name of the option is not very suggestive.  From the code I see that we want to decrease the value of `hotBorderFrequency` for compilations at warm or below, under serverInlining and for methods that are very small or always worth inlining (`calleeResolvedMethod->maxBytecodeIndex() < comp()->getOptions()->getAlwaysWorthInliningThreshold() || alwaysWorthInlining(calleeResolvedMethod, callNode)`). Maybe the option name should capture that.
Interestingly `coldBorderFrequency` is increased from 50 to 500 meaning that more blocks could be considered cold than in the past.

Reply:
Update in force push.